lib/cabal.nix

```
@@ -0,0 +1,12 @@
+{ lib, nixpkgs, get-tested }: let
+  pkgs = nixpkgs.legacyPackages.x86_64-linux;
+in {
+  # This function extracts GHC versions in the format used by haskell.nix
+  # from a "tested-with" stanza in the cabal file
+  getGhcVersions = cabalFile: let
```

A tiny nitpick, I think this way the name is a bit more descriptive of where the versions are coming from
```suggestion
  getTestedWithVersions = cabalFile: let
```